### PR TITLE
hide two classes and one enumeration in `Azure.ResourceManager`

### LINKED ISSUE #29294: Hide `KeyvaultProperties` and `EncryptionProperties` in ResourceManager
These 2 models should not be in Azure.ResourceManager lib, need to be marded as EBNever and remove `PropertyReferenceType` tag.

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/29294

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/resourcemanager/Azure.ResourceManager/src/autorest.md [hunk 1]
```diff
@@ -14,8 +14,8 @@ mgmt-debug:
   show-request-path: true
 batch:
   - tag: package-common-type-2022-04
-  - tag: package-resources-2022-04
-  - tag: package-management-2022-04
+#   - tag: package-resources-2022-04
```
Comment (ArthurMa1978) [inline sdk/resourcemanager/Azure.ResourceManager/src/autorest.md:17]: consider to remove comment out lines

File: sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/EncryptionProperties.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.Models
+{
+    // this class here is to keep the class public and add the EditorBrowsableNever attribute
+    // this class is exposed in resourcemanager by accident, now we hide it in resourcemanager
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public partial class EncryptionProperties
+    {
```
Comment (ArthurMa1978) [inline sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/EncryptionProperties.cs:14]: If empty, then my understanding is a breaking change

File: sdk/resourcemanager/Azure.ResourceManager/src/autorest.md (hunk 1 above)
Comment (ArcturusZhang) [reply sdk/resourcemanager/Azure.ResourceManager/src/autorest.md:17]: Sorry, I forgot this. This is only to accelerate my local generation

File: sdk/resourcemanager/Azure.ResourceManager/src/Common/Generated/Models/KeyVaultProperties.cs [hunk 3]
```diff
@@ -5,24 +5,19 @@
 
 #nullable disable
 
-using Azure.Core;
-
 namespace Azure.ResourceManager.Models
 {
     /// <summary> The KeyVaultProperties. </summary>
-    [PropertyReferenceType]
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager/src/Common/Generated/Models/KeyVaultProperties.cs:13]: Lets leave these attributes in place and add a provision in autorest.csharp to ignore reference types marked with Obsolete

File: sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/EncryptionProperties.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+#nullable disable
+
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.Models
+{
+    // this class here is to keep the class public and add the EditorBrowsableNever attribute
+    // this class is exposed in resourcemanager by accident, now we hide it in resourcemanager
+    [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/EncryptionProperties.cs:12]: Add the Obsolete as warning attribute here as well.
`[Obsolete("This type is obsolete and will be removed in a future release.", false)]`

**Review by m-nash — CHANGES_REQUESTED** [review]

File: sdk/resourcemanager/Azure.ResourceManager/src/autorest.md [hunk 5]
```diff
@@ -52,56 +52,90 @@ rename-rules:
...
+      $["x-ms-mgmt-propertyReferenceType"] = true;
+      $["x-namespace"] = "Azure.ResourceManager.Models";
+      $["x-accessibility"] = "public";
+      $["x-csharp-formats"] = "json";
+      $["x-csharp-usage"] = "model,input,output";
   - from: types.json
     where: $.definitions.systemData.properties.*
     transform: >
-      $["readOnly"] = true
+      $["readOnly"] = true;
```
Comment (fengzhou-msft) [inline sdk/resourcemanager/Azure.ResourceManager/src/autorest.md:100]: I think we can update the `x-csharp-usage` to be `model,output`, then we can remove the below workaround to set each property in `systemData` as readOnly.

**Review by m-nash — APPROVED** [review]
Approved from Apiview perspective

File: sdk/resourcemanager/Azure.ResourceManager/src/autorest.md (hunk 5 above)
Comment (ArcturusZhang) [reply sdk/resourcemanager/Azure.ResourceManager/src/autorest.md]: I will have a try

Comment (ArcturusZhang) [reply sdk/resourcemanager/Azure.ResourceManager/src/autorest.md]: well, it works, but it also changes a constructor from public to internal, which is a breaking change. I would like to keep the previous implementation...

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**